### ISSUE TITLE
GetFrame still working / how to convert Picture to JPEG?

### ISSUE THREAD

**Issue by bknill** [jcodec/jcodec#107]
FrameGrab.getFrame is depreciated and replaced with FrameGrab.getNativeFrame() I assume, which returns a Picture. 

If this is what I'm supposed to use to convert a frame into an image, how do I convert a Picture into JPEG?

**Comment by svitvitskiy** [comment]
No, GetNativeFrame should also be deprecated. Use FrameGrab8Bit

**Comment by bknill** [comment]
Thanks sorry beginner question how do I import that? 

org.jcodec.api.FrameGrab8Bit isn't there?

**Comment by bknill** [comment]
Ignore I found it!

**Comment by bknill** [comment]
Opened from source, but  FrameGrab8Bit still returns a Picture8Bit - how do I get a JPEG from that?

**Comment by bknill** [comment]
Does JCodec actually still do the first function in the example (extract frames from movie) ?

**Comment by svitvitskiy** [comment]
You'd have to use FrameGrab from javase project.

jcodec/javase/src/main/java/org/jcodec/api/awt/FrameGrab8Bit.java

**Comment by bknill** [comment]
Great thanks!

**Comment by bknill** [comment]
![scene_1671_pano_1](https://cloud.githubusercontent.com/assets/7074483/11911990/7f63df76-a68e-11e5-8b31-4c2dd3e3152a.jpg)

I've managed to make some beautiful art like this. It's not quite what I had in mind. Any idea what that might be about?

**Comment by svitvitskiy** [comment]
I think your chroma channels are oversaturated.

This might be a bug in JCodec. And you are just using FrameGrab?

Stan

**Comment by bknill** [comment]
Yea just using FrameGrab8Bit

            BufferedImage frame = FrameGrab8Bit.getFrame(file, 10);
            System.out.println(frame.getColorModel());

            File tempFile = new File(Constants.TEMP_DIR_PATH + file.getName() + ".jpg");
            ImageIO.write(convertedImg, "jpg",frame);

Is it worth me using an older version before FrameGrab got depriciated?

**Comment by svitvitskiy** [comment]
I'd say at least try it.

If the result is different, please file a bug I'll fix it.

Stan

**Comment by bknill** [comment]
Do you know which was the last version before FrameGrab got depriciated and where I can download the Java SE version?

**Comment by svitvitskiy** [comment]
It's marked as deprecated to discourage use but it still works in the
current version.

Stan

**Comment by bknill** [comment]
The server compilation is having problems with the depreciated version,  shall I open a new issue for the colour problems in the new version?

**Comment by svitvitskiy** [comment]
Yes please.

**Comment by toddjmorgan** [comment]
I'm getting the same results as bknill...but only for videos taken on an android phone.  videos taking on iPhones give good pics.

**Comment by svitvitskiy** [comment]
Could you please post some video samples so I could reproduce?

**Comment by svitvitskiy** [comment]
Please try it now, I fixed some of the color problems in the last couple of commits.
